Mathlib/Analysis/Cauchy.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weber
+-/
+import Mathlib.Algebra.EuclideanDomain.Basic
+import Mathlib.Algebra.EuclideanDomain.Field
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Degree.Lemmas
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Data.Complex.Abs
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ :=
+  if h : p.natDegree = 0 then
+    0
+  else
+    sup' (range p.natDegree) (by simpa) (fun x ↦ ‖p.coeff x‖) / ‖p.leadingCoeff‖
```

Comment:
What are your thoughts on
```suggestion
noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
  sup (range p.natDegree) (fun x ↦ ‖p.coeff x‖₊) / ‖p.leadingCoeff‖₊
```
which avoids the `if` and the `sup'`?

Reply:
I tried switching to `NNReal`, I think it's an improvement, except for `NNReal.geom_sum`, which should be moved to a different file. Do you know of a good place for it?